### ISSUE TITLE
Error in importation of linopy : MOSEK license not found

### ISSUE THREAD

**Issue by robingirard** [PyPSA/linopy#206]
Dear Linopy community, I have installed version 0.3.1. I'm happy that MOSEK is now a possible solver, but when I import linopy on a computer without mosek I have the following error :

In [7]: import linopy
Traceback (most recent call last):
  File "/home/rgirard/.virtualenvs/LEAP/lib/python3.10/site-packages/IPython/core/interactiveshell.py", line 3548, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-7-4b79a7083b10>", line 1, in <module>
    import linopy
  File "/home/rgirard/.virtualenvs/LEAP/lib/python3.10/site-packages/linopy/__init__.py", line 12, in <module>
    from linopy import model, remote
  File "/home/rgirard/.virtualenvs/LEAP/lib/python3.10/site-packages/linopy/model.py", line 21, in <module>
    from linopy import solvers
  File "/home/rgirard/.virtualenvs/LEAP/lib/python3.10/site-packages/linopy/solvers.py", line 61, in <module>
    m.optimize()
  File "/home/rgirard/.virtualenvs/LEAP/lib/python3.10/site-packages/mosek/__init__.py", line 10781, in optimize
    return self.__optimizetrm__1(*args,**kwds)
  File "/home/rgirard/.virtualenvs/LEAP/lib/python3.10/site-packages/mosek/__init__.py", line 10770, in __optimizetrm__1
    raise Error(rescode(_res_optimizetrm),_msg_optimizetrm)
mosek.Error: rescode.err_missing_license_file(1008): License cannot be located. The default search path is ':/home/rgirard/mosek/mosek.lic:'.

It should be possible to import linopy without having a mosek license ! Am I missing something ?

**Comment by fneum** [comment]
You're totally right! No MOSEK license should be required.

@FabianHofmann something maybe going wrong with the new context manager?

https://github.com/PyPSA/linopy/blob/a97febee4ecf206d94e9acb15cea2b7efff7094a/linopy/solvers.py#L57-L66

**Comment by FabianHofmann** [comment]
mmh, it works for me for all combinations:

- package uninstalled, no license
- package installed, no license
- package installed, with license

are you on the latest master?

**Comment by robingirard** [comment]
I installed with pip.

**Comment by FabianHofmann** [comment]
okay, could you then update the linopy package via pip? I just released version 0.3.2

**Comment by FabianHofmann** [comment]
great!

**Comment by asre244** [comment]
Hello @FabianHofmann

I am having a similar issue where I am not able to use the MOSEK solver although I do not get any error messages. I have installed the MOSEK package in the environment but it does not show up in the list of "available_solvers" in "linopy". I tried updating "linopy" as explained above but does not seem to change anything. Interestingly, I was able to use MOSEK a couple of days ago before the weekend but when I tried to use it again, it did not show up in the list of "available_solvers". I tried to work with gurobi to check if I was missing something but I was able to use gurobi. 

![available_solvers](https://github.com/PyPSA/linopy/assets/105757079/30df2250-6d35-4137-aba1-55c3ce6b0a6e)
![installed packages](https://github.com/PyPSA/linopy/assets/105757079/da6e7e8a-e396-4923-a786-416b71c166a4)
